dive_core/tests/string_utils_test.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ Copyright 2025 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+#include "dive_core/common/string_utils.h"
+#include "gtest/gtest.h"
+#include <limits>
+
+namespace Dive
+{
+namespace StringUtils
+{
+
+TEST(StringUtils, SafeConvertFromString_Int)
+{
+    int val = 0;
+    EXPECT_TRUE(SafeConvertFromString("123", val));
+    EXPECT_EQ(val, 123);
+
+    EXPECT_TRUE(SafeConvertFromString("-123", val));
+    EXPECT_EQ(val, -123);
+
+    EXPECT_TRUE(SafeConvertFromString("0", val));
+    EXPECT_EQ(val, 0);
+
+    EXPECT_FALSE(SafeConvertFromString("123a", val));
+    EXPECT_FALSE(SafeConvertFromString("abc", val));
+    EXPECT_FALSE(SafeConvertFromString("", val));
+    EXPECT_FALSE(SafeConvertFromString("123.45", val));
+
+    // Test limits
+    EXPECT_TRUE(SafeConvertFromString(std::to_string(std::numeric_limits<int>::max()), val));
+    EXPECT_EQ(val, std::numeric_limits<int>::max());
+    EXPECT_TRUE(SafeConvertFromString(std::to_string(std::numeric_limits<int>::min()), val));
+    EXPECT_EQ(val, std::numeric_limits<int>::min());
+
+    // Out of range
+    long long too_big = (long long)std::numeric_limits<int>::max() + 1;
```

Comment:
Prefer named casts over C casts since the intent is more explicit and the compiler can catch errors. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast 
```suggestion
    long long too_big = static_cast<long long>(std::numeric_limits<int>::max()) + 1;
```